backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/sponsoredbanner/internal/BpkSponsoredBannerImpl.kt

```diff
@@ -0,0 +1,193 @@
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.sponsoredbanner.internal
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.widthIn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.sponsoredbanner.BpkSponsoredBannerCTA
+import net.skyscanner.backpack.compose.sponsoredbanner.BpkSponsoredBannerVariant
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.InformationCircle
+import net.skyscanner.backpack.compose.utils.clickable
+
+@Composable
+internal fun BpkSponsoredBannerImpl(
+    backgroundColor: Color,
+    variant: BpkSponsoredBannerVariant,
+    title: String?,
+    subHeadline: String?,
+    callToAction: BpkSponsoredBannerCTA?,
+    body: String?,
+    modifier: Modifier = Modifier,
+    logo: @Composable (() -> Unit)?,
+) {
+    val showBody = !body.isNullOrBlank()
+    var isBodyVisible by remember { mutableStateOf(false) }
+    val toggleShowBody = {
+        isBodyVisible = !isBodyVisible
+    }
+
+    Column(
+        modifier = modifier
+            .background(
+                color = BpkTheme.colors.canvasContrast,
+                shape = RoundedCornerShape(BpkSpacing.Md),
+            ),
+    ) {
+        BpkSponsoredBannerHeader(
+            modifier = Modifier
+                .background(
+                    color = backgroundColor,
+                    shape = if (isBodyVisible) {
+                        RoundedCornerShape(
+                            topStart = BpkSpacing.Md,
+                            topEnd = BpkSpacing.Md,
+                            bottomStart = BpkSpacing.None,
+                            bottomEnd = BpkSpacing.None,
+                        )
+                    } else {
+                        RoundedCornerShape(BpkSpacing.Md)
+                    },
+                )
+                .clickable(enabled = showBody, onClick = toggleShowBody),
```

Comment:
NIT: the current ripple effect goes outside of rounded corners. You can resolve it by applying clip modifier before the clickable or specifying the radius parameter for rememberRipple() call inside clickable function (you'll need to pass it though)